packages/expo-modules-core/ios/AppDelegates/ExpoAppDelegate.swift

```diff
@@ -2,6 +2,10 @@ import Dispatch
 import Foundation
 import React_RCTAppDelegate
 
+#if canImport(ReactAppDependencyProvider)
```

Comment:
```suggestion
// TODO(vonovak,20250107) - Remove the if expression when we drop SDK 52 support
#if canImport(ReactAppDependencyProvider)
```

would be good to add a TODO comment as a reminder to cleanup the code

great idea to use canImport here, which is a clever change 👏

Reply:
added the comment